states/etcd_backup.go

```diff
@@ -89,7 +89,9 @@ func getBackupEtcdCmd(cli *clientv3.Client, basePath string) *cobra.Command {
 // backupEtcd backup all key-values with prefix provided into local file.
 // implements gzip compression for now.
 func backupEtcd(cli *clientv3.Client, base, prefix string, component string, filePath string) error {
-	resp, err := cli.Get(context.Background(), path.Join(base, prefix), clientv3.WithCountOnly(), clientv3.WithPrefix())
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*3)
+	resp, err := cli.Get(ctx, path.Join(base, prefix), clientv3.WithCountOnly(), clientv3.WithPrefix())
+	cancel()
```

Comment:
It's recommended to use `defer cancel()` in go practice.

Reply:
fixed.